src/components/Progress.js

```diff
@@ -34,35 +28,28 @@ export default class Progress extends React.PureComponent {
   }
 
   render() {
-    let innerStyle
-    switch (this.props.active) {
-      case 2:
-        innerStyle = { width: '100%' }
-        break
-      case 1:
-        innerStyle = { animation: `${this.state.duration}ms linear 0ms ${style.slidein}`, animationPlayState: this.props.pause ? 'paused' : 'running' }
-        break
-      case 0:
-        innerStyle = { width: 0 }
-        break
-      default:
-        innerStyle = { width: 0 }
-        break
-    }
+    const { width, pause, bufferAction, active, defaultInterval } = this.props
     return (
-      <div className={style.autoHide} style={{...styles.progress, ...{width: `${this.props.width * 100}%`, opacity: this.props.pause && !this.props.bufferAction ? 0 : 1}}}>
-        <div ref={r => { this.inner = r }} className={style.inner} style={innerStyle} /* style={{...styles.overlay, width: `${this.props.completed * 100}%`}} */ />
-      </div>
+      <ProgressWrapper width={width} pause={pause} bufferAction={bufferAction}>
+        <div ref={r => { this.inner = r }} className={style.inner} style={getProgressStyle({ active, defaultInterval, pause })} />
```

Comment:
instead of the `defaultInterval` you need to pass `this.state.duration`

Reply:
also, in the functions below (`animation`, `getProgressStyle`), I'd rather rename the argument `defaultInterval` to `duration`, it makes more sense

Reply:
@ovionlogis Thanks for the feedback. I've just updated it.